src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -34,10 +36,46 @@ public static void UpdateAutomationId(this AView nativeView, IView view)
 		{
 			if (AutomationTagId == DefaultAutomationTagId)
 			{
-				AutomationTagId = Microsoft.Maui.Resource.Id.automation_tag_id;
+				AutomationTagId = Resource.Id.automation_tag_id;
 			}
 
 			nativeView.SetTag(AutomationTagId, view.AutomationId);
 		}
+
+		public static void UpdateShadow(this AView nativeView, IView view)
+		{
+			var radius = view.Shadow.Radius;
+
+			if (radius < 0)
+				return;
+
+			var opacity = view.Shadow.Opacity;
+
+			if (opacity < 0)
+				return;
+
+			var nativeColor = view.Shadow.Color.ToNative();
+
+			if (nativeView is TextView textView)
+			{
+				var offsetX = (float)view.Shadow.Offset.Width;
+				var offsetY = (float)view.Shadow.Offset.Height;
+				textView.SetShadowLayer(radius, offsetX, offsetY, nativeColor);
+				return;
+			}
+
+			nativeView.OutlineProvider = view.BackgroundColor.A > 0
+				? ViewOutlineProvider.PaddedBounds
+				: ViewOutlineProvider.Bounds;
+
+			if (nativeView.Context != null)
+				nativeView.Elevation = nativeView.Context.ToPixels(radius);
+
+			if (!NativeVersion.Supports(NativeApis.OutlineAmbientShadowColor))
+				return;
```

Comment:
What should we do in lower versions?. We can implement Shadows on lower versions but Android draws shadow on a software layer. So the whole view must be drawn on a software layer, which means no hardware acceleration at all. The best option are those API 28 methods.